src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,176 @@
-""" This is a dummy example """
+import requests
+import pandas as pd
+from pandas import DataFrame
+import plotly.express as px  # type: ignore
+import streamlit as st
+import time
+from typing import Any, Dict
+
+API_URL = "https://archive-api.open-meteo.com/v1/archive"
+COORDINATES = {
+    "Madrid": {"latitude": 40.416775, "longitude": -3.703790},
+    "London": {"latitude": 51.507351, "longitude": -0.127758},
+    "Rio": {"latitude": -22.906847, "longitude": -43.172896},
+}
+VARIABLES = ["temperature_2m", "precipitation", "wind_speed_10m"]
+
+
+PLOT_INFO = {
+    "temperature_2m": {"color": "red", "y_label": "Mean Temperature (ºC)"},
+    "precipitation": {"color": "blue", "y_label": "Accumulated precipitation (mm)"},
+    "wind_speed_10m": {"color": "green", "y_label": "Max. Wind speed 10 meters (km/h)"},
+}
+
+
+def get_data_meteo_api(city: str, magnitude: str) -> Dict[str, Any]:
+    params = {
+        "latitude": str(COORDINATES[city]["latitude"]),
+        "longitude": str(COORDINATES[city]["longitude"]),
+        "start_date": "2010-01-01",
+        "end_date": "2019-12-31",
+        "hourly": magnitude,
+    }
+
+    r = requests.get(API_URL, params=params)
+    print(r.status_code)
+    data = r.json()
```

Comment:
This will probably fail as soon as the request GET fails and status code isn't 200. When we work with external dependencies we have to be very defensive in our programming and try to account for as many potential issues as possible, so we can gratefully handle and fail on those scenarios.